packages/aws-cdk-lib/aws-rds/lib/cluster.ts

```diff
@@ -793,9 +831,14 @@ abstract class DatabaseClusterNew extends DatabaseClusterBase {
    * Perform validations on the cluster instances
    */
   private validateClusterInstances(writer: IAuroraClusterInstance, readers: IAuroraClusterInstance[]): void {
+    let performanceInsightsDuplicate = false;
```

Comment:
Shouldn't we validate that `performanceInsightEncryptionKey` and `performanceInsightRetention` match between the cluster and instances instead? We're blocking valid deployments with this setup.

> If the value of enablePerformanceInsights was different for the cluster and the instance, it did not result in an error. If it was true in the cluster and false in the instance, it did not result in an error but was also enabled in the instance.

Should we display a warning for this scenario?

Reply:
I added the validations for `performanceInsightEncryptionKey` and `performanceInsightRetention` and the warning . The implementation has become a bit complicated, what do you think?